connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/messaging/mqtt/hivemq/subscribing/MqttSubscriber.java

```diff
@@ -120,10 +123,25 @@ private Source<SubscribeResult, NotUsed> subscribe(final GenericMqttSubscribe ge
     private SubscribeResult consumeIncomingPublishesForSubscribedTopics(
             final org.eclipse.ditto.connectivity.model.Source connectionSource
     ) {
+        final List<MqttTopicFilter> topicFilters =
+                connectionSource.getAddresses().stream().map(MqttTopicFilter::of).toList();
```

Comment:
Why making a list out of the stream when it gets right afterwards converted to a stream again?

Reply:
Because a stream can be consumed only once.

Reply:
:+1: